rcon/player_history.py

```diff
@@ -289,9 +329,72 @@ def remove_player_from_blacklist(steam_id_64):
             raise CommandFailedError(f"Player {player} was not blacklisted")
 
         player.blacklist.is_blacklisted = False
+
+        if rcon:
+            _find_and_remove_perma_bans_by_ban_log(rcon, player.ban_logs, sess)
+
         sess.commit()
 
 
+def remove_player_from_blacklist_using_ban_log(ban_log):
+    with enter_session() as sess:
+        players = sess.query(
+          PlayerSteamID
+        ).join(
+          PlayerSteamID.ban_logs
+        ).filter(
+          PlayerBanLog.ban_log == ban_log
+        ).all()
+
+        if not players:
+            return
+
+        for player in players:
+            remove_player_from_blacklist(player.steam_id_64)
+
+
+def _get_players_who_are_not_blacklisted_but_have_ban_log(sess):
+    return sess.query(
+        PlayerSteamID
+    ).join(
+        PlayerSteamID.ban_logs
+    ).join(
+        PlayerSteamID.blacklist
+    ).filter(
+        PlayerBanLog.ban_log != None
+    ).filter(
+        BlacklistedPlayer.is_blacklisted == False
+    ).all()
+
+
+def _banned_unblacklisted_players(rcon, sess):
+    return_list = []
+    players = _get_players_who_are_not_blacklisted_but_have_ban_log(sess)
+
+    perma_bans = rcon.get_perma_bans()
+    for player in players:
+        for ban_log in [ban_log.ban_log for ban_log in player.ban_logs]:
+            if ban_log in perma_bans:
+                return_list.append(player)
+                break
+
+    return return_list
+
+
+def banned_unblacklisted_players(rcon):
+    with enter_session() as sess:
+        return [p.to_dict() for p in _banned_unblacklisted_players(rcon, sess)]
+
+
+def unban_unblacklisted_players(rcon):
```

Comment:
This file is getting quite length, maybe it would be worth extracting all the blacklist related code to a new file?